qiskit/dagcircuit/dagdependency.py

```diff
@@ -533,12 +533,12 @@ def _does_commute(node1, node2):
     """
 
     # Create set of qubits on which the operation acts
-    qarg1 = [node1.qargs[i].index for i in range(0, len(node1.qargs))]
-    qarg2 = [node2.qargs[i].index for i in range(0, len(node2.qargs))]
+    qarg1 = [node1.qargs[i] for i in range(0, len(node1.qargs))]
+    qarg2 = [node2.qargs[i] for i in range(0, len(node2.qargs))]
 
     # Create set of cbits on which the operation acts
-    carg1 = [node1.qargs[i].index for i in range(0, len(node1.cargs))]
-    carg2 = [node2.qargs[i].index for i in range(0, len(node2.cargs))]
+    carg1 = [node1.cargs[i] for i in range(0, len(node1.cargs))]
+    carg2 = [node2.cargs[i] for i in range(0, len(node2.cargs))]
```

Comment:
Wow, that's surprising nobody caught that we were putting qubits there until now.

Reply:
It seems like you'd hit this only if you were looking to run `TemplateOptimization` on a circuit with more than one `qreg` (but if you did, you'd hit it right away). Might be an edge case no one had run into yet, since its fairly new, and not in the preset pass managers.